brave/brave-core
# Remember a user's contact information that they provide in a webcompat report and auto-fill it

Resolves https://github.com/brave/brave-browser/issues/40021



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

1. Install the Brave browser
2. Open Settings -> Brave Shields & privacy
3. Toggle switch: `Store contact information for future webcompat reports` must be ON (as default )
4. Open new Tab and URL address you would like to report
5. Open Brave Shields pop up dialog
6. Switch off Brave Shields if It is ON
7. Click "Report broken site" 
8. Enter any contact information
9. Click "Submit"
10. Go through points 5-6-7 and make sure that previously entered contact info **appears** in the dialog field, and has been sent in the report content.
11. Go through points 4-5-6-7 but for Private Tab, make sure that previously entered contact info **does not appear** in the dialog field, and has been sent in the report content.
12. Open Settings -> Brave Shields & privacy
13. Change the state of the `Store contact information for future webcompat reports` to OFF
14. Return to normal Tab (not private)
15. Go through points 5-6-7 and make sure that previously entered contact info **does not appear** in the dialog field
16. Enter some contact info
17. Click "Submit" 
18. Make sure that entered contact info has been sent in the report content
19. Go through points 5-6-7 and make sure that previously entered contact info **does not appear** in the dialog field
